ddtrace/profiling/collector/_lock.py

```diff
@@ -185,8 +185,10 @@ def _release(self, inner_func, *args, **kwargs):
             # _self_acquired_at is only set when the acquire was captured
             # if it's not set, we're not capturing the release
             start = self._self_acquired_at
```

Comment:
This could fail too if another thread (erroneously) unlocks the lock at the same time and hits `del` between the `hasattr` check and the `self._self_acquired_at` access, right?

Reply:
Updated code and added comments.